examples/metrics_simple/metrics_ostream.cc

```diff
@@ -44,21 +44,49 @@ void initMetrics(const std::string &name)
       new metric_sdk::MeterProvider(std::move(exporters)));
   auto p = std::static_pointer_cast<metric_sdk::MeterProvider>(provider);
   p->AddMetricReader(std::move(reader));
+
+  // counter view
   std::unique_ptr<metric_sdk::InstrumentSelector> instrument_selector{
       new metric_sdk::InstrumentSelector(metric_sdk::InstrumentType::kCounter, name)};
   std::unique_ptr<metric_sdk::MeterSelector> meter_selector{
       new metric_sdk::MeterSelector(name, version, schema)};
-  std::unique_ptr<metric_sdk::View> view{
+  std::unique_ptr<metric_sdk::View> sum_view{
       new metric_sdk::View{name, "description", metric_sdk::AggregationType::kSum}};
-  p->AddView(std::move(instrument_selector), std::move(meter_selector), std::move(view));
+  p->AddView(std::move(instrument_selector), std::move(meter_selector), std::move(sum_view));
+
+  // histogram view
+  std::unique_ptr<metric_sdk::InstrumentSelector> histogram_instrument_selector{
+      new metric_sdk::InstrumentSelector(metric_sdk::InstrumentType::kHistogram, name)};
+  std::unique_ptr<metric_sdk::MeterSelector> histogram_meter_selector{
+      new metric_sdk::MeterSelector(name, version, schema)};
+  std::unique_ptr<metric_sdk::View> histogram_view{
+      new metric_sdk::View{name, "description", metric_sdk::AggregationType::kHistogram}};
+  p->AddView(std::move(histogram_instrument_selector), std::move(histogram_meter_selector),
+             std::move(histogram_view));
   metrics_api::Provider::SetMeterProvider(provider);
 }
 }  // namespace
-int main()
+
+int main(int argc, char **argv)
 {
+  if (argc != 2)
+  {
+    std::puts("invalid number of command line arguments");
+    return 0;
+  }
+
+  std::string example_type(argv[1]);
   std::string name{"ostream_metric_example"};
   initMetrics(name);
-  foo_library(name);
+
+  if (example_type == "counter")
+  {
+    foo_library::counter_example(name);
+  }
+  else
+  {
+    foo_library::histogram_example(name);
+  }
```

Comment:
Would also be a good use-case to create both counter and histogram if no argument is specified - to demonstrate getting metrics from multiple instruments simultaneously.

Reply:
thanks, added multiple instruments simultaneously.